src/System.Net.Http/tests/StressTests/HttpStress/Program.cs

```diff
@@ -500,6 +521,17 @@ void ValidateContent(string expectedContent, string actualContent)
                         // Echos back the requested content in a full duplex manner.
                         await context.Request.Body.CopyToAsync(context.Response.Body);
                     });
+                    endpoints.MapPost("/speed", async context =>
+                    {
+                        var buffer = new byte[serverBufferSize];
+                        int bytesRead = await context.Request.Body.ReadAsync(buffer);
+
+                        while (bytesRead != 0)
+                        {
+                            await context.Response.Body.WriteAsync(buffer, 0, bytesRead);
+                            bytesRead = await context.Request.Body.ReadAsync(buffer);
+                        }
```

Comment:
This is a strange way to write this loop.  Something like:
```C#
int bytesRead;
while ((bytesRead = await context.Request.Body.ReadAsync(buffer)) > 0)
{
    await context.Response.Body.WriteAsync(buffer, 0, bytesRead);
}
```
is more typical, or if you prefer:
```C#
while (true)
{
    int bytesRead = await context.Request.Body.ReadAsync(buffer);
    if (bytesRead == 0) break;
    await context.Response.Body.WriteAsync(buffer, 0, bytesRead);
}
```
but either way, we don't need multiple call sites to ReadAsync.